raiden/network/transport/matrix/transport.py

```diff
@@ -254,13 +254,13 @@ def __repr__(self):
 class MatrixTransport(TransportLayer, Runnable):
     _room_prefix = "raiden"
     _room_sep = "_"
-    log = log
 
     def __init__(self, address: Address, config: dict, current_server_name: str = None):
         TransportLayer.__init__(self, address)
         Runnable.__init__(self)
         self._config = config
         self._raiden_service: Optional[RaidenService] = None
+        self.log = log
 
```

Comment:
not entirely sure what's going on here.

for starters, there is no apparent reason why `log` should be a _class_ attribute rather than an instance attribute. even moreso when it's just pointing to the variable on line `79`.

for some reason, if both:
1. `log` remains a class attribute
2. `MatrixTransport` implements multiple inheritance (`TransportLayer` in addition to the previously present `Runnable`)

then `MatrixTransport` **turns into an abstract class.** 

this seems baffling, but i've made sure it is the case by calling `MatrixTransport.__abstractmethods__` and verifying that `log` appears to be an abstract method. since the class has an abstract method, it cannot be instantiated, and therefore transport objects won't come to be, and the whole node will come crashing down in flames.

what's even more mystifying: if `log` is commented out—or, in this case, turned into an instance variable—then the problem apparently disappears. 

inspecting the `log` variable in runtime points to some very obscure types which i'm not familiar with, and don't really understand how they can lead into an abstract method being part of this class once we have multiple inheritance.

this issue came to light after running the test suite. it still needs more looking into.

Reply:
this is probably related to a new failure in `integration/api/test_restapi.py`:
```shell
___________________________________________________ test_api_payments_target_error[matrix-False-2] ___________________________________________________

api_server_test_instance = <raiden.api.rest.APIServer object at 0x7efd5caf6fd0>, raiden_network = [<App 007eb599>, <App f0feeae8>]
token_addresses = [b'\x19\xac\xdf\xba\x08\x1b\xc7\x15\xe39@\x0b\x9b\xdf(\xb0\xe6\xd5\xf4\x06']

    @pytest.mark.parametrize("number_of_nodes", [2])
    def test_api_payments_target_error(api_server_test_instance, raiden_network, token_addresses):
        _, app1 = raiden_network
        amount = 200
        identifier = 42
        token_address = token_addresses[0]
        target_address = app1.raiden.address
    
        # stop app1 to force an error
        app1.stop()
    
        request = grequests.post(
            api_url_for(
                api_server_test_instance,
                "token_target_paymentresource",
                token_address=to_checksum_address(token_address),
                target_address=to_checksum_address(target_address),
            ),
            json={"amount": amount, "identifier": identifier},
        )
        response = request.send().response
        assert_proper_response(response, status_code=HTTPStatus.CONFLICT)
>       app1.start()

integration/api/test_restapi.py:812: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../app.py:150: in start
    self.raiden.start()
../raiden_service.py:477: in start
    self._initialize_whitelists(chain_state)
../raiden_service.py:1008: in _initialize_whitelists
    self._set_hub_transport_whitelist(chain_state)
../raiden_service.py:992: in _set_hub_transport_whitelist
    self.transport.hub_transport.whitelist(neighbour)
../network/transport/matrix/transport.py:461: in whitelist
    self.log.debug("Whitelist", address=to_normalized_address(address))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MatrixTransport node:f0feeae8 id:139626653468544>, item = 'log'

    def __getattr__(self, item: str) -> Any:
>       return getattr(self.greenlet, item)
E       AttributeError: 'gevent._greenlet.Greenlet' object has no attribute 'log'

../utils/runnable.py:78: AttributeError

```

Reply:
i found the [root cause](https://github.com/hynek/structlog/issues/229) of the problem.

this is the fix that i've pushed: 49c357338990e226900e9fb22631c5dc42077277